framework/src/play-akka-http-server/src/main/scala/play/core/server/akkahttp/AkkaModelConversion.scala

```diff
@@ -216,45 +253,80 @@ private[server] class AkkaModelConversion(
     }
   }
 
-  private def convertHeaders(headers: Iterable[(String, String)]): immutable.Seq[HttpHeader] = {
-    headers.map {
-      case (name, value) =>
-        HttpHeader.parse(name, value) match {
-          case HttpHeader.ParsingResult.Ok(header, errors /* errors are ignored if Ok */ ) =>
-            header
-          case HttpHeader.ParsingResult.Error(error) =>
-            sys.error(s"Error parsing header: $error")
-        }
-    }.to[immutable.Seq]
+  private def convertHeaders(headers: Iterable[(String, String)]): immutable.Seq[HttpHeader] =
+    headers.collect {
+      case (name, value) if name != TRANSFER_ENCODING =>
+        RawHeader(name, value)
+    }(collection.breakOut): Vector[HttpHeader]
+}
+
+final case class AkkaRequestHeaders(
+    request: HttpRequest,
+    uri: String,
+    override val contentLength: Long,
+    contentType: Option[String],
+    hs: immutable.Seq[HttpHeader],
+    isChunked: Boolean
+) extends Headers(null) {
```

Comment:
It's probably OK to change the `Headers` superclass into a `trait` if you like. It is technically an API change in our public APIs but I think it's OK since not many people will be implementing their own custom header logic.